### DISCUSSION TITLE
Remove APKTOOL_DUMMY

### DISCUSSION THREAD

**Discussion by pashamcr** [iBotPeaches/Apktool#3159] [Ideas]
**Is your feature request related to a problem? Please describe.**
Apktool creates a lot of unnecessary APKTOOL_DUMMY, which means nothing but empty space, but in the output file they take up space as dead weight.

**Describe the solution you'd like**
Delete or comment out the code chunks that cause APKTOOL_DUMMY.

**Additional context**
For example, apk https://www.apkmirror.com/apk/google-inc/gboard/gboard-10-9-03-387080390-release/gboard-the-google-keyboard-10-9-03-387080390-lite_release-arm64-v8a-android-apk-download/
`aapt2 dump resources Gboard.apk > res.txt`
for three apks: original, re-compiled, and re-compiled using apktool without APKTOOL_DUMMY.
Compare logs res.txt which one is closer to the original? In addition, the output file size increases
: original 33.0 mb, output with APKTOOL_DUMMY 33.4 mb, without APKTOOL_DUMMY 33.1 mb
[res.txt](https://github.com/iBotPeaches/Apktool/files/7365855/res.txt)
[res_dummy.txt](https://github.com/iBotPeaches/Apktool/files/7365856/res_dummy.txt)
[res_nodummy.txt](https://github.com/iBotPeaches/Apktool/files/7365858/res_nodummy.txt)

**Comment by iBotPeaches** [comment]
We tried this [once](https://github.com/iBotPeaches/Apktool/pull/2463) and I was hesitant, maybe for 3.0. Marking as feature.

**Comment by iBotPeaches** [comment]
I'm closing this as resolved. We had issues in past with dummy attributes because of not handling sparse resources properly, so we would pad many resources that simply aren't needed.

Once that was resolved during 2.8.0/2.8.1 - the only time dummy resources would come into play I see near impossible now. If we skip resources that are missing (NO_ENTRY) after reading all sparse resources. There would probably never be one anymore.

See this 2.8.2 test against the original application in this.

```
➜  dist aapt2 d resources 3159.apk| grep -i 'apktool'
➜  dist 
```

No apktool resources after decode/build. Closing as completed.
